presto-orc/src/main/java/com/facebook/presto/orc/reader/LongStreamReader.java

```diff
@@ -38,11 +39,11 @@
     private final LongDictionaryStreamReader dictionaryReader;
     private StreamReader currentReader;
 
-    public LongStreamReader(StreamDescriptor streamDescriptor)
+    public LongStreamReader(StreamDescriptor streamDescriptor, LocalMemoryContext systemMemoryContext)
     {
         this.streamDescriptor = requireNonNull(streamDescriptor, "stream is null");
-        directReader = new LongDirectStreamReader(streamDescriptor);
-        dictionaryReader = new LongDictionaryStreamReader(streamDescriptor);
+        directReader = new LongDirectStreamReader(streamDescriptor, systemMemoryContext);
```

Comment:
I am not sure this is correct. For example, say `directReader` allocated 10 bytes, and the `dictionaryReader` allocated 20 bytes. Since both readers do a `setBytes` we will track either 10 or 20 bytes depending on who calls `setBytes` the last, however we should track `30` bytes. I think using an `AggregatedMemoryContext` here will properly track that. You can pass `systemMemoryContext.newAggregatedMemoryContext()` in `StreamReaders` to this constructor and add a comment here why we make an exception for this stream reader.